### ISSUE TITLE
Doesn't remember window size 

### ISSUE THREAD

**Issue by thany** [vector-im/element-web#20940]
### Steps to reproduce

1. Start Element and set the window size how you like
2. Close Element
3. Start Element again

#### What did you expect?
It remember windows size & configuration exactly the way I closed it.

#### What happened instead?
The main window is a lot smaller when it starts, and when I resize it back the way I way, the sidebar also expands which I have to collapse manually.

### Operating system

Windows 10 21H2

### Application version

Element version: 1.10.1 Olm version: 3.2.8

### Will you send logs?

No

**Comment by t3chguy** [comment]
Defect because we call upon `electron-window-state` to handle this for us, but it hasn't been updated in 3 years and may have some bugs

**Comment by thany** [comment]
If it's not been updated, how come it suddenly started having this problem? This is a regression, mind you. Previous version did not have this problem.

Something must've been changed in order for this bug to surface.

**Comment by t3chguy** [comment]
> If it's not been updated, how come it suddenly started having this problem?

Electron has been updated, so the interaction between Electron and some code which relies on Electron behaving in a specific way will change, even if only half of that whole has changed.

**Comment by t3chguy** [comment]
> So it has been updated. I thought you wrote it hadn't been. Okay.

No, `electron-window-state` hasn't been updated, but Electron has.
Just like if you update from Windows 10 to 11 some software can stop working the same as it used to.

**Comment by thany** [comment]
Okay, fair enough. But let's start thinking about possibilities, rather than impossibilities.

If `electron-window-state` is the culprit, replace it, insource it, fix it, or take a different approach. Maybe look at what solution other apps have chosen to remember window size&position. VScode is very mature, and is opensource. Maybe look thereabouts.

**Comment by thany** [comment]
Seeing that this is a regression, has this issue gotten any headway yet? It's kind of really important that the very start of a program (literally starting it up) does get onto a user's nerves. Also, regressions should get elevated priority.

**Comment by claell** [comment]
Just ran into this. It is especially bad on a multi monitor setup.

**Comment by thany** [comment]
It's really disappointing not to have a solution after 6 months. I'm sure you're busy, but this stands out to me twice a day, and any additional times I have to do a reboot.

I cannot, in good conscience, recommend this program to friends & colleagues, beause they too all have multimonitor setups (on which apparently this is more obvious - I wouldn't know, it's *always* obvious to me). And a bug that feels like it should be so damn easy to fix (don't know if it is, but it *feels* that way) is a real letdown for firsttimers, because they **will** encounter this bug and go "like, what, have they ever opened this program themselves? are they not SEEING this?", I would imagine.

But I digress. Since this issue is so _bleedingly obvious_, and it's been sitting for half a year, please elevate priority.

**Comment by t3chguy** [comment]
> And a bug that feels like it should be so damn easy to fix

You're welcome to try, the power of open source and all that.

We're currently using https://www.npmjs.com/package/electron-window-state but it seems abandoned and has outstanding issues.

**Comment by thany** [comment]
> You're welcome to try, the power of open source and all that.

The default opensource way to brush off an issue. Because everyone who reports an issue, is magically a programmer and had knowledge of the exact packages and technologies used.

Anyway, had priority been raised? Has anyone done anything to look into it? I don't have to re-iterate how absolutely monstrously obvious this bug is, do it? You know, the fact that *every single time* the app starts, it starts with a window in the exact right location, but 2-3x too small? Yeah that one, that problem that presumably everyone with two different monitors is having. So basically everyone with, say, any laptop and any external monitor hooked up.

Maybe not rely on external packages to do something so seemingly benign as remembering X, Y, Width, and height of your application window. I don't know, but it seems pretty damn basic to me.

**Comment by t3chguy** [comment]
> I don't know, but it seems pretty damn basic to me.

If it was basic then the package probably wouldn't be buggy, no? It has ~25k weekly downloads. 

Lots of apps implement it themselves with buggy behaviour - anecdotes and open issues linked from https://github.com/electron/electron/issues/526

**Comment by thany** [comment]
My setup doesn't change between starts. It should be as trivial as remembering X/Y/H/W.

**Comment by thany** [comment]
Since this issue reproduces 100% of the time on multiple computers, I shouldn't think this would be too hard to get reproduced...

So, do you need more information to solve this? Is there something you don't understand yet? Do you need help, or pointers, or anything at all?

One thing I personally don't understand, is how this bug is allowed to linger on for many releases, even though it is so blatantly obviously there. Doesn't it like, super obviously stand out to anyone else, literally every time the program starts?

**Comment by t3chguy** [comment]
To fix this personally I'd need a swathe of environments to test it on whilst developing, handful of different Linux DEs specifically. This isn't something I would have time to go off-piste on. We rely on a library to handle this for us to act as the abstraction layer between all the various types of environments, obviously the library has issues of its own. There are a lot of possible edge cases to handle here, this simply isn't as trivial as one would think.

**Comment by thany** [comment]
I'm waiting for questions then. What do you need from me?

**Comment by t3chguy** [comment]
Nothing? This issue doesn't have an `X-Needs-Info` label?

**Comment by thany** [comment]
Then why are you telling me basically "work on my machine"? That's just adding to my frustration. That's always adding to a reporter's frustration. Wouldn't you be annoying when you reporting something broken and they're telling you "nope, works fine here"?

Anyway, good that it works for you. Keep it that way. Don't break it on your own machine while fixing it, I guess.

**Comment by soonic6** [comment]
Version von Element: 1.11.28
Version von Olm: 3.2.12

Edition	Windows 11 Pro
Version	22H2
Installiert am	‎21.‎09.‎2022
Betriebssystembuild	22621.1413
Leistung	Windows Feature Experience Pack 1000.22639.1000.0


will not remeber windows size, position and monitor.
